rasa_addons/core/tracker_stores/botfront.py

```diff
@@ -38,134 +69,119 @@ def _start_sweeper(tracker_store, break_time):
-        except Exception as e:
-            logger.error(
-                "Error: post request to botfront-api failed with error: {}".format(e)
-            )
+    def _insert_tracker_gql(self, sender_id, tracker):
+        data = self._graphql_query(
+            INSERT_TRACKER,
+            {"senderId": sender_id, "projectId": self.project_id, "tracker": tracker},
+        )
+        return data.get("insertTrackerStore")
 
-    @staticmethod
-    def _is_none(tracker):
-        return tracker is None or (
-            isinstance(tracker, UnsynchronizedTracker) and not tracker.is_initialized()
+    def _update_tracker_gql(self, sender_id, tracker):
+        data = self._graphql_query(
+            UPDATE_TRACKER,
+            {"senderId": sender_id, "projectId": self.project_id, "tracker": tracker},
         )
+        return data.get("updateTrackerStore")
+
+    def _get_last_index(self, sender_id):
+        info = self.trackers_info.get(sender_id, -1)
+        if info == -1:
+            return info
+        elif info.get("last_index") is None:
+            return -1
+        else:
+            return info.get("last_index")
+
+    def _get_last_timestamp(self, sender_id):
+        info = self.trackers_info.get(sender_id, 0)
+        if info == 0:
+            return info
+        elif info.get("last_timestamp") is None:
+            return 0
+        else:
+            return info.get("last_timestamp")
+
+    def _store_tracker_info(self, sender_id, tracker_info):
+        self.trackers_info[sender_id] = {
+            "last_index": tracker_info["lastIndex"],
+            "last_timestamp": tracker_info["lastTimestamp"],
+        }
 
     def save(self, canonical_tracker):
 
         # Fetch here just in case retrieve wasn't called first
         sender_id = canonical_tracker.sender_id
-        tracker = self._fetch_tracker(sender_id)
+        tracker = self.trackers.get(sender_id)
         serialized_tracker = self._serialize_tracker_to_dict(canonical_tracker)
 
-        if self._is_none(tracker):
-            # No document in the collection, so insert one
-
-            if not isinstance(tracker, UnsynchronizedTracker):
-                logger.debug(
-                    "{} -> INSERTING TRACKER: {}".format(sender_id, serialized_tracker)
-                )
-                self._post(sender_id, "insert", serialized_tracker)
-                self.trackers[sender_id] = serialized_tracker
-            else:
-                logger.debug("{} -> UNSYNCHRONIZED TRACKER UPDATE".format(sender_id))
-                self.trackers[sender_id] = UnsynchronizedTracker(serialized_tracker)
-
+        if tracker is None:  # the tracker does not exist localy ( first save)
+            updated_info = self._insert_tracker_gql(sender_id, serialized_tracker)
+            self.trackers[sender_id] = serialized_tracker
+            # update the last index and last time stamp for future uses
+            self._store_tracker_info(sender_id, updated_info)
```

Comment:
only if update_info is not None

Reply:
I'have added the check in _store_tracker_info